Build cache - Various small adjustments

gsmet opened the pull request:
By adding an output timestamp, we can get it to be binary identical when the content is identical.
Obviously, the date needs to be forged to be consistent between two separate builds and the output to be cachable.

gsmet committed:
Make quarkus-ide-launcher jar reproducible

By adding an output timestamp, we can get it to be binary identical when
the content is identical.
Obviously, the date needs to be forged to be consistent between two
separate builds and the output to be cachable.

gsmet commented:
@aloubyansky pinging you as I'm not sure it's safe? But I don't think we are ever using the timestamp of the files in this jar?

maxandersen commented:
i'm missing sometihng - where is there a timestamp missing for ide-launcher? I thought removing timestamps wold be the need?

maxandersen commented:
ah scratch that - the PR view was missingfiles.

maxandersen commented on core/launcher/pom.xml:
```diff
@@ -92,6 +92,7 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
                 <configuration>
+                    <outputTimestamp>2023-10-17T10:15:30Z</outputTimestamp>
```
normally, one would use the date from the git sha or a 1970 date ...
does this specific timestamp have any meaning?

gsmet replied:
It's today and that's pretty much it :).

Using the date from the git sha is a bit counter productive as that means that it will be different for each git commit, which might not be needed.

I can go with 1970 if you prefer but really we don't really care about this date.

gsmet committed:
Build cache - Ignore Eclipse local files in spotless input

gsmet committed:
Build cache - Ignore quarkus-ide-launcher in classpath

It shouldn't be around anyway as only useful for IDE runs so let's get
it out of the classpath.

aloubyansky commented:
Could you please explain why it doesn't work w/o this change?

gsmet commented:
@aloubyansky I should have mentioned that it's related to the issue I pinged you for in Zulip: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/IDE.20launcher.20in.20classpath.20when.20running.20tests.20from.20console .

As this dependency is in the classpath of the ITs, if the jar is different, then it's a cache miss thus why I tried to make it reproducible, which seems to work with this option (and it's what they recommend in the doc).

I then also tried https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/IDE.20launcher.20in.20classpath.20when.20running.20tests.20from.20console which might help.

gsmet commented:
Merged as I need this in to make progress on the Build cache. If you want me to adjust something, please create an issue and I will have a look.
